consensus-types/helpers/comparisons.go

```diff
@@ -0,0 +1,109 @@
+package helpers
```

Comment:
Sorry, we need tests for all of these :scream: . Maybe a friendly AI could help with some of that work?